### PR TITLE
Task: Abstract out generation of Person string

### LINKED ISSUE #152: Task: Abstract out generation of Person string
The generation of the Person string shown to the user is duplicated across the various commands. Let's abstract this to the `Person` class to reduce code duplication and improve maintainability.

### PR DESCRIPTION
**Pull request by dillontkh**
Resolves #152

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public class EditCommand extends Command {
     private final EditPersonDescriptor editPersonDescriptor;
 
     /**
-     * @param index of the person in the filtered person list to edit
+     * @param index                of the person in the filtered person list to edit
```
Comment (bgopi23) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:62]: on github it looks like there is an extra space here? is it the same for u?

Comment (bgopi23) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:62]:
```suggestion
     * @param index of the person in the filtered person list to edit
```

Comment (dillontkh) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:62]: I think it was due to IntelliJ's formatting to try and keep the descriptions of the params in line. Good catch!

**Review by bgopi23 — APPROVED** [review]
LGTM! thanks for the abstractions

**Review by ruijietay — APPROVED** [review]
Great catch! Code looks more abstracted now 😃

**Pull request merged**